### PR TITLE
People wearing kitty ears can have their tails pulled

### PR DESCRIPTION
**Pull request by Thunder12345**
## About The Pull Request

Extends tail pulling to people wearing kitty ears. If you pull on their fake tail it will tip off and give you a mood debuff for ripping someone's tail off.

Added a new sound cloth_rip.ogg, a shortened version of dimensional_rend.ogg with one ripping noise.

## Why It's Good For The Game

Pulling people's tails is funny, so it's even funnier when it comes off in your hands.

## Changelog

:cl:
add: People wearing kitty ears can now have their tails pulled.
/:cl:

### CODE REVIEW

**Comment by Jackraxxus** [comment]
I was going to ask if this applied to cyborgs wearing kitty ears but you specified "people".
Unlucky I guess.

**Comment by timothymtorres** [comment]
> I was going to ask if this applied to cyborgs wearing kitty ears but you specified "people". Unlucky I guess.

Cyborgs don't get mood effects.

**Comment by msgerbs** [comment]
> > I was going to ask if this applied to cyborgs wearing kitty ears but you specified "people". Unlucky I guess.
> 
> Cyborgs don't get mood effects.

It's not the pullee who gets the mood debuff, it's the puller.

**Comment by Time-Green** [comment]
> > > I was going to ask if this applied to cyborgs wearing kitty ears but you specified "people". Unlucky I guess.
> > 
> > 
> > Cyborgs don't get mood effects.
> 
> It's not the pullee who gets the mood debuff, it's the puller.

👍  retracted. I want to mutilate degenerates and have my character enjoy it as much as I do

**Comment by CPTANT** [comment]
A 5 point mood debuff for pulling away a cosmetic item is ridiculous.

**Comment by IndieanaJones** [comment]
Should be a mood debuff to the person who loses the tail to punish people who wear cat ears

**Comment by Farquaar** [comment]
Why would I get a mood debuff for removing a furry's fake tail but not get a debuff for violently torturing and murdering people?

File: code/modules/mob/living/carbon/carbon_defense.dm [hunk 1]
```diff
@@ -437,6 +437,17 @@
 		else
 			add_mood_event("tailpulled", /datum/mood_event/tailpulled)
 
+	else if ((helper.zone_selected == BODY_ZONE_PRECISE_GROIN) && (istype(head, /obj/item/clothing/head/costume/kitty) || istype(head, /obj/item/clothing/head/collectable/kitty)))
```
Comment (ZephyrTFA) [inline code/modules/mob/living/carbon/carbon_defense.dm:440]: Technically `else if` is invalid syntax that Byond helpfully interprets correctly anyway, would it be possible for you to also refactor this proc to use early returns instead?
And by invalid syntax I mean that
```
else if
```
Is invalid while
```
else
	if
```
is valid

Comment (timothymtorres) [reply code/modules/mob/living/carbon/carbon_defense.dm:440]: `else if` is valid syntax and it's used extensively in the codebase.

Code_DVEomzuufA

Comment (ZephyrTFA) [reply code/modules/mob/living/carbon/carbon_defense.dm:440]: > `else if` is valid syntax and it's used extensively in the codebase.
> 
> Code_DVEomzuufA

According to Ike it's not valid syntax it's just that the compiler is lenient about it

Comment (msgerbs) [reply code/modules/mob/living/carbon/carbon_defense.dm:440]: It's literally in the DM docs: https://www.byond.com/docs/ref/#/proc/if

Comment (timothymtorres) [reply code/modules/mob/living/carbon/carbon_defense.dm:440]: > According to Ike it's not valid syntax it's just that the compiler is lenient about it

Who is Ike? `else if` loops are super common and any person working with our codebase should be familiar with them.

Comment (ZephyrTFA) [reply code/modules/mob/living/carbon/carbon_defense.dm:440]: > > According to Ike it's not valid syntax it's just that the compiler is lenient about it
> 
> Who is Ike? `else if` loops are super common and any person working with our codebase should be familiar with them.

Main person coding OpenDream basically

Comment (ZephyrTFA) [reply code/modules/mob/living/carbon/carbon_defense.dm:440]: Regardless, I still think that this proc should have early returns in it and not be a chiant if else chain

Comment (tralezab) [reply code/modules/mob/living/carbon/carbon_defense.dm:440]: i'm fixing that

**Review by tralezab — APPROVED** [review]
bad implementation but i'm just going to fix it in my pr

**Pull request merged**